### PR TITLE
Refactor with BattleContext and Pokemon classes

### LINKED ISSUE #32: Refactor code to use BattleContext class
Now that we have these helper classes, we need to actually use them.

All the functions that use battle context as dicts that need to be converted to use battle context as `BattleContext`
- [ ] `battleapi.get_battle_context`
- [ ] `util.prompt_for_turn`

### LINKED ISSUE #34: Refactor code to use Pokemon class
Now that we have these helper classes, we need to actually use them.

All the functions that use pokemon as dicts that need to be converted to use pokemon as `Pokemon`
- [ ] `BattleContext.__init__`
- [ ] `battleapi,generate_pokemon`
- [ ] `util.prompt_for_turn`

### PR DESCRIPTION
**Pull request by A-very-Cunning-ham**
Refactor code to use BattleContext and Pokemon classes

Closes #32, Closes #34

### CODE REVIEW

File: battleapi.py [hunk 1]
```diff
@@ -44,7 +45,8 @@ async def get_battle_context(battle_id: int, target: int) -> dict:
 	async with aiohttp.ClientSession() as session:
 		async with session.get(f"{BASE_URL}/battle/context?id={battle_id}&target={target}") as resp:
 			result = await resp.json()
-	return result
+			battlecontext = BattleContext(**result)
+	return battlecontext
```
Comment (dyc3) [inline battleapi.py:49]:
```suggestion
	return BattleContext(**result)
```

File: pkmntypes.py [hunk 2]
```diff
@@ -53,7 +53,7 @@ def __init__(self, **kwargs):
 		self.EVs: list[int] = kwargs['EVs']
 		self.Nature: int = kwargs['Nature']
 		self.Stats: list[int] = kwargs['Stats']
-		self.StatusModifiers: list[int] = kwargs['StatusModifiers']
+		self.StatusModifiers: list[int] = kwargs['StatModifiers'] # TODO: clarify if this should read Stat or Status
```
Comment (dyc3) [inline pkmntypes.py:56]: It should be `StatModifiers`
```suggestion
		self.StatModifiers: list[int] = kwargs['StatModifiers']
```

File: util.py [hunk 3]
```diff
@@ -8,14 +8,14 @@
 
 RESPONSE_REACTIONS = ["🇦", "🇧", "🇨", "🇩"]
 
-async def prompt_for_turn(bot: commands.Bot, user: discord.User, battlecontext) -> Turn:
+async def prompt_for_turn(bot: commands.Bot, user: discord.User, BattleContext) -> Turn:
```
Comment (dyc3) [inline util.py:11]: avoid name conflict
```suggestion
async def prompt_for_turn(bot: commands.Bot, user: discord.User, battlecontext: BattleContext) -> Turn:
```

**Review by dyc3 — CHANGES_REQUESTED** [review]

File: util.py (hunk 3 above)
Comment (dyc3) [reply util.py:11]: this will also require the other references to this variable to be changed

Comment (A-very-Cunning-ham) [reply util.py]: got it. thanks

**Review by dyc3 — APPROVED** [review]

**Pull request merged**